tests/test_perf.py

```diff
@@ -67,6 +68,8 @@ def test_perf_fp_dwarf_smart(
         assert is_function_in_collapsed(";_start;__libc_start_main;main;", process_collapsed) ^ bool(
             perf_mode not in ("dwarf", "smart")
         )
+        # Check if container name is added to ProfileData
+        assert application_docker_container.name == snapshot_pid_profile(profiler, application_pid).container_name
```

Comment:
Avoid calling `snapshot` again - use `snapshot_pid_profiler` in line 54, then compare both the container name & the collapsed of it. Calling `snapshot` again invokes the profiler twice, unnecessarily.